### PR TITLE
Support for Rails Credentials in addition to Env Vars

### PR DESCRIPTION
**Pull request by benhayehudi**
This PR introduces the ability to instantiate a Nexmo client in a Rails app by either credentials or environment variables. It conditions the Nexmo client config values based on the presence of  `Rails.application.credentials.nexmo`.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
 # 0.1.0
 
-* First version!
\ No newline at end of file
+* First version!
+
+# 0.2.0
```
Comment (fabianrbz) [inline CHANGES.md]: Nitpick: Usually the newest changes are at the top, so you don't need to scroll all the way down in order to see the latest changes.

File: nexmo_rails.gemspec [hunk 2]
```diff
@@ -21,6 +21,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency('dotenv-rails')
   spec.add_development_dependency('rspec')
   spec.add_development_dependency('generator_spec')
+  spec.add_development_dependency('rails-dummy')
```
Comment (fabianrbz) [inline nexmo_rails.gemspec]: @benhayehudi do we need this dependency?

File: lib/generators/nexmo_initializer/nexmo_initializer_generator.rb [hunk 3]
```diff
@@ -1,11 +1,34 @@
...
+      end
+
+      env_initializer
+    end
+
+    private
+
+    def credentials_initializer
+      initializer "nexmo.rb" do <<~HEREDOC
+        Nexmo.setup do |config|
```
Comment (fabianrbz) [inline lib/generators/nexmo_initializer/nexmo_initializer_generator.rb]: nitpick: I prefer to have the `=` aligned :p, but you can leave it as it is

**Comment by benhayehudi** [comment]
> @benhayehudi i left some comments, mostly nitpicks.
> It looks good 💪 .
> 
> The only thing is that CI is not set up for this gem so I had to manually run the specs to be sure they were passing. I would say that we can tackle that as part of a different issue, but it's definitely something that we should address. WDYT?

Thank for the lookover! I addressed the issues you raised. I left the `=` sign as is since it might bug me to keep on making sure they were all aligned all the time. :) 

Setting up the CI for the gem I think is a good next step to implement to enable better processes for future work on it.

**Comment by benhayehudi** [comment]
Thanks @fabianrbz -- I'm going to hold off on merging for a little bit, so going to add a `don't merge` label for now. Appreciate the review!

**Pull request merged**